program/src/token.rs

```diff
@@ -51,7 +51,7 @@ pub type Result<T> = std::result::Result<T, ArithmeticError>;
 /// only used for `Debug` and `Display` printing.
 #[macro_export]
 macro_rules! impl_token {
-    ($TokenLamports:ident, $symbol:expr) => {
+    ($TokenLamports:ident, $symbol:expr, $decimals:expr) => {
```

Comment:
To make the call site a bit more obvious, we might write
```rust
($TokenLamports:ident, $symbol:expr, decimals = $decimals:expr) => {
```
or even
```rust
(type = $TokenLamports:ident, symbol = $symbol:expr, decimals = $decimals:expr) => {
```
and then at the call site it would become
```
impl_token!(UstLamports, "UST", decimals = 6);
```
so the meaning of the magic number is immediately clear.

Reply:
Great, let's go for the describing the `decimals` magic number. The rest is more intuitive.